hack/fake-apiserver/README.md

```diff
@@ -0,0 +1,46 @@
+# CAPM3 Fake API server
+
+Fake API server is a tool running inside a kubernetes cluster,
+and "generates" in-memory k8s api server endpoints on demand.
+These endpoints can be used, for e.g., to represent the clusters
+created by CAPM3 using fake hardware so that CAPI confirms that
+the cluster was provisioned successfully.
+
+## How to use
+
+After building the container image and deploy it to a kubernetes cluster,
+you can generate a fake API server endpoint by sending
+a GET request to the fake API server, with the following params:
+
+```shell
+"namespace": namespace in which the cluster will be created
+"cluster_name": name of the cluster
+"caKey": generated CA key
+"caCert": generated CA cert
+"etcdKey": generated etcd key
+"etcdCert": generated etcd cert
+```
+
+```shell
+curl <pod_ip>:<port>/register?resource=<namespace>/<cluster_name>&caKey=<caKeyEncoded>&caCert=<caCertEncoded>&etcdKey=<etcdKeyEncoded>&etcdCert=<etcdCertEncoded>"
+```
+
+The fake API server will return a response with the ip and port of the newly
+generated api server. This can be fed to a CAPI infrastructure provider
+(for e.g. CAPM3) to create a cluster.
+
+After the cluster is created, additional information can be added to the
```

Comment:
What additional information? If I understand the code correctly, it is for adding nodes and nothing else. If so, can you just write that instead so it is clear what this can be used for